src/rgw/rgw_asio_frontend.cc

```diff
@@ -165,8 +166,10 @@ void handle_connection(boost::asio::io_context& context,
                                     &real_client))));
       RGWRestfulIO client(cct, &real_client_io);
       auto y = optional_yield{context, yield};
-      process_request(env.store, env.rest, &req, env.uri_prefix,
-                      *env.auth_registry, &client, env.olog, y, scheduler);
+      ret_code = process_request(env.store, env.rest, &req, env.uri_prefix,
+                                 *env.auth_registry, &client, env.olog, y, scheduler);
+      if (ret_code == -ERR_INCOMPLETE_WRITE_REQ)
+	continue;
```

Comment:
this check should come after the `if (!parser.keep_alive()) { return; }` so that we respect `Connection: close` headers